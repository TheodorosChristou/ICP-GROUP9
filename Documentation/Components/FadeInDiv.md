# FadeInDiv Component

This component creates a fading animation effect using CSS keyframes.

## Props

- `children`: ReactNode - The content to be wrapped by the fading animation effect.

## Example Usage

```tsx
import FadeInDiv from './FadeInDiv';

const YourComponent = () => {
  return (
    <div>
      <h1>Your Website</h1>
      <FadeInDiv>
        <p>This content will fade in when rendered.</p>
      </FadeInDiv>
    </div>
  );
};

export default YourComponent;
```

## Component Structure
```tsx
<FadeInDiv>
  <div className="..."></div>
</FadeInDiv>
```

## Implementation Details

- Uses keyframes from @emotion/react to define the fading animation.

- Uses styled from @emotion/styled to create a styled component with the fading animation.

- Renders children components wrapped by the fading animation effect.

```tsx
import React from 'react';
import { keyframes } from '@emotion/react';
import styled from '@emotion/styled';

// Define the fadeIn animation
const fadeIn = keyframes`
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
`;

// Styled component with the fadeIn animation
const FadingDiv = styled.div`
  animation: ${fadeIn} 1.5s ease-in-out;
`;

interface FadeInDivProps {
  children: React.ReactNode;
}

const FadeInDiv: React.FC<FadeInDivProps> = ({ children }) => {
  return (
    <FadingDiv>
      {children}
    </FadingDiv>
  );
};

export default FadeInDiv;
```