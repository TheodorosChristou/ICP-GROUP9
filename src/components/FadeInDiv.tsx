import React from 'react';
import { keyframes } from '@emotion/react';
import styled from '@emotion/styled';

// Define the fadeIn animation
const fadeIn = keyframes`
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
`;

// Styled component with the fadeIn animation
const FadingDiv = styled.div`
  animation: ${fadeIn} 1.5s ease-in-out;
`;

interface FadeInDivProps {
  children: React.ReactNode;
}

const FadeInDiv: React.FC<FadeInDivProps> = ({ children }) => {
  return (
    <FadingDiv>
      {children}
    </FadingDiv>
  );
};

export default FadeInDiv;