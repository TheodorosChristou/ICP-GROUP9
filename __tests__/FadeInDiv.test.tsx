import React from 'react'
import {render} from '@testing-library/react'
import FadeInDiv from '@/components/FadeInDiv'; 

describe('FadeInDiv component', () => {
  it('renders children', () => {
    // Define some children to render within the FadeInDiv component
    const children = <div data-testid="child">Child Component</div>;

    // Render the FadeInDiv component with the children
    const { getByTestId } = render(<FadeInDiv>{children}</FadeInDiv>);

    // Assert that the children are rendered
    expect(getByTestId('child')).toBeInTheDocument();
  });


  it('matches snapshot', () => {
    // Render the FadeInDiv component
    const { container } = render(<FadeInDiv>Child Component</FadeInDiv>);

    // Assert that the rendered component matches the snapshot
    expect(container).toMatchSnapshot();
  });
});
