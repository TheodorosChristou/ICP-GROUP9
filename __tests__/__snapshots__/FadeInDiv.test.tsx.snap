// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeInDiv component matches snapshot 1`] = `
<div>
  <div
    class="css-11apktd"
  >
    Child Component
  </div>
</div>
`;
